nomenclature/codelist.py

```diff
@@ -639,7 +640,7 @@ def from_directory(
         mapping: Dict[str, RegionCode] = {}
         for code in code_list:
             if code.name in mapping:
-                raise DuplicateCodeError(name=name, code=code.name)
+                raise ValueError(f"Trying to set a duplicate code {code.name}")
```

Comment:
Is this reversion to a ValueError done on purpose here?

Reply:
Yes it is, this was actually a left over from the move to pydantic v2. In v2 pydantic does not really allow you to create custom errors anymore so I removed them all. This one slipped through the cracks also because there was no test for it. When working on #309 I noticed that but forgot to address it there.
If you want I can bring back the `DuplicateCodeError` as a custom error. This time though it would be derived from a `ValueError` and not from a pydantic error.

Reply:
Ok, thanks!